src/ORM/DataQuery.php

```diff
@@ -505,15 +516,20 @@ public function exists(): bool
         }
 
         // Wrap the whole thing in an "EXISTS"
-        $sql = 'SELECT CASE WHEN EXISTS(' . $statement->sql($params) . ') THEN 1 ELSE 0 END';
-        $result = $this->withCorrectDatabase(
-            fn() => DB::prepared_query($sql, $params)
+        $subQuerySql = $statement->sql($params);
+        $selectExists = SQLSelect::create('1')->setUseCache($this->useCache, $this->dataClass())->addWhere(['EXISTS (' . $subQuerySql . ')' => $params]);
+
+        $queryResult = $this->withCorrectDatabase(
+            fn() => $selectExists->execute()
         );
-        $row = $result->record();
-        $result = reset($row);
+        $row = $queryResult->record();
+        if ($row) {
+            $result = reset($row);
+        } else {
+            $result = false;
+        }
```

Comment:
The result is now empty if the query is returning false, instead of `0` that it used to return.